### DISCUSSION TITLE
Execute commands with sudo

### DISCUSSION THREAD

**Discussion by takasurazeem** [community/community#25981] [Actions]
<p>Hi, I have a GitHub action set on self hosted Ubuntu VPS on Linode.<br>
I ran <code>sudo ./svc.sh install</code> and then <code>./svc.sh start</code>. The runner is up and working but I have a command in my yaml file that needs to run as sudo. I have this command in sudoers file to run without sudo<br>
<code>%sudo ALL= NOPASSWD: /bin/systemctl restart twentydb.service </code><br>
But still when I push on the server the following error occurs.<br>
<code>sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper</code></p>
<blockquote>
<ul>
<li>name: restart the service<br>
run: |<br>
sudo systemctl restart twentydb</li>
</ul>
</blockquote>
<p>Please help, thanks. I also need to setup 2 more runners on the same server and same setup.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The command in your workflow file is not the same as the one in your <code>sudoers</code> file, <code>sudo</code> doesn’t know they are functionally equivalent. Use <code>twentydb.service</code> in the workflow, too.</p>
<p>Also, is the user your runner runs as in the <code>sudo</code> group? If not, that entry won’t apply to it. However I’d suggest using a separate entry for the user rather than adding the runner to the <code>sudo</code> group.</p>

**Comment by takasurazeem** [comment]
<p>Yes, thanks for pointing that out, everything works as expected now, is it permissible to run two more runners in the same account and VPS? Also</p>
<blockquote>
<p>However, I’d suggest using a separate entry for the user rather than adding the runner to the sudo group.</p>
</blockquote>
<p>I didn’t get that advice, could you please elaborate a little?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="takasurazeem" data-post="3" data-topic="121264">
<div class="title">
<div class="quote-controls"></div>
 takasurazeem:</div>
<blockquote>
<p>I didn’t get that advice, could you please elaborate a little?</p>
</blockquote>
</aside>
<p>It’s a security rule to give each process as little rights as possible, to minimize the damage in case of compromise or bugs. On most systems adding the runner to the <code>sudo</code> group will also give it general sudo rights (with password), and the rule above will give any other user in the group the right to restart the <code>twentydb</code> service without giving a password. This is not the most critical example for that rule (though that depends on the setup of your VPS), but it’s still a good idea to follow it. :slightly_smiling_face:</p>
<p>Regarding the questions about runner details I hope someone with experience on that matter can chip in.</p>
